### DISCUSSION TITLE
Use of ColumnSet for setting up "SELECT" or "RETURNING" clauses

### DISCUSSION THREAD

**Discussion by sebastorama** [vitaly-t/pg-promise#870] [Q&A] [answered]
Given that I have the following ColumnSet:

```typescript
    const returnCS = new pgp.helpers.ColumnSet([
      'ingredientId',
      'recipeId',
      {
        name: 'inclusionRatio',
        cast: 'float',
      },
      {
        name: 'pricePerKg',
        cast: 'float',
      },
    ]);
```

The important bit being the `cast: 'float'` part on the 2 numeric fields. Is there a way to use this inside a SELECT or RETURNING clause to produce something like this: `"ingredientId", "recipeId", "inclusionRatio"::float, "pricePerKg"::float` ?

using "returnCS.names"  and the name filter will produce the column fields, but without the `::float` bit. Is ColumnSet intended more for insert/update statements?

**Comment by vitaly-t** [comment, marked as answer]
`ColumnSet` is intended for the functions provided within the `helpers` namespace. "Select" isn't there, and not meant to be.

If you want to reuse `ColumnSet` in that manner, you can create a function that will generate such syntax:

```ts
function listColumns(cs : ColumnSet) : string {
    return cs.columns.map(c => c.escapedName + (c.cast ? '::' + c.cast : '')).join(', ');
}
```

**Reply by sebastorama** [reply]
Thanks a lot for your work and support!

Sidetracking, but adjacent to this, I noticed that numeric fields come to JS land as strings (probably the pg driver itself does that, floats comes as number). Is there a more elegant way of getting that as numbers? I thought of doing "parseFloats" on the returned objects.

(The thinking behind this probably is due to possible loss of precision on an IEEE 754 representation of floats, so the user is making that decision knowingly the possible side-effects)

**Comment by vitaly-t** [comment]
> I noticed that numeric fields come to JS land as strings (probably the pg driver itself does that, floats comes as number). Is there a more elegant way of getting that as numbers? I thought of doing "parseFloats" on the returned objects.

Types conversion is managed by [pg-types](https://github.com/brianc/node-pg-types), which you can access via `pgp.pg.types`. See also [this question](https://stackoverflow.com/questions/39168501/pg-promise-returns-integers-as-strings).
